src/test/java/tools/jackson/databind/jsontype/ExistingPropertyWithSealedTypesTest.java

```diff
@@ -0,0 +1,546 @@
+package tools.jackson.databind.jsontype;
+
+import java.util.*;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.As;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+
+import tools.jackson.core.type.TypeReference;
+import tools.jackson.databind.ObjectMapper;
+import tools.jackson.databind.testutil.DatabindTestUtil;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ExistingPropertyWithSealedTypesTest extends DatabindTestUtil
```

Comment:
On naming: let's use common prefix ("SealedTypesWith...") ending with "...Test" for these, to group test classes next to each other.

Older pattern of "TestXxx" is being slowly replaced by "...Test" but let's not add new classes using that (but leave `TestSubtypes` as-is for now, rename can be done with separate PR).